src/ray/protobuf/gcs.proto

```diff
@@ -270,6 +272,13 @@ message ActorCheckpointIdData {
   repeated uint64 timestamps = 3;
 }
 
+message WorkerFailureData {
+  // Address of the worker that failed.
+  Address worker_address = 1;
```

Comment:
This must include `WorkerID` too to avoid the port reuse issue, right? IIRC this can be included in `Address` but is optional/not always present. It might be better to have a separate `WorkerAddress` type for clarity (and then we can use that for the intended `WorkerID` checks).

Reply:
Or we could only include WorkerID and omit the other details? Not sure why you would want to connect to a dead worker.

Reply:
I thought about just using the WorkerID but thought it might be nice to include the address information in case we want to display information about where the failures occurred later on (e.g., through the web UI).

Reply:
Seems fine; consider adding a check that the id is present.